presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -167,6 +167,23 @@ public Response createQuery(
                     .build());
         }
 
+        // The Teradata Presto ODBC Driver checks node version to decide the Presto's statement
+        // protocol and sends test queries about PREPARE statement.
+        // Rewrite the statement so that Presto always returns version for the compatible protocol.
+        // Ban the statement which will never be consumed by the driver.
+        if (servletRequest.getHeader("User-Agent").equals("Teradata Presto ODBC Driver")) {
```

Comment:
Would be great to put these strings as static final string somewhere at the beginning of this class